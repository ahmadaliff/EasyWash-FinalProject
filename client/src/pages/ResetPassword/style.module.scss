@import '@styles/base';

.mainWrap {
  @include directionX;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  width: 100vw;
  max-width: 100%;
  .logoHeader {
    position: absolute;
    cursor: pointer;
    width: auto;
    top: 0;
    padding: 1rem;
    height: 2.5rem;
    .logo {
      height: 2.5rem;
    }
  }
  .forgotPasswordCard {
    @include directionY;
    justify-content: center;
    align-items: center;
    @include apply-at-md-size {
      width: 100%;
      padding: 5rem;
    }
    width: 100%;
    padding: 2rem;
    background-color: var(--color-bg-secondary);
    .forgotPasswordHeader {
      width: 100%;
      text-align: center;
    }
    .form {
      .buttonSubmit {
        width: 100%;
        background-color: var(--color-primary);
        color: var(--color-bg);
        border-radius: 2.5rem;
        padding: 0.5rem 2.5rem 0.5rem 2.5rem;
        margin: auto;
        margin-top: 0.75rem;
        box-shadow: none;
        border: none;
        cursor: pointer;
        &:hover {
          transform: scale(1.0125, 1.0125);
        }
        &:disabled {
          background-color: rgba(gray, 0.2);
          color: rgba($color: gray, $alpha: 0.5);
        }
        &:disabled:hover {
          transform: none;
        }
      }
    }
    .showPassword {
      @include directionX;
      align-items: center;
      padding-top: 0.25rem;
      font-size: small;
    }
  }
  .formNav {
    @include directionY;
    width: 100%;
    padding-top: 1rem;
    gap: 0.25rem;
    .nav {
      margin: 0;
      font-size: small;
      a {
        color: var(--color-primary);
      }
    }
  }
}
