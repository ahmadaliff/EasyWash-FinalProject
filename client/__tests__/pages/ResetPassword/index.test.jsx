import { render, fireEvent, screen, act } from '@utils/testHelper';
import ResetPassword from '@pages/ResetPassword';

let wrapper;
const mockDispatch = jest.fn();
const mockNavigate = jest.fn();
const mockUseParams = jest.fn().mockReturnValue({ token: 'token' });

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
  useParams: () => mockUseParams,
  Link: () => <b />,
}));
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}));

beforeEach(() => {
  act(() => {
    wrapper = render(<ResetPassword />);
  });
  jest.clearAllMocks();
});

describe('ResetPassword Page', () => {
  describe('with login', () => {
    jest.mock('@containers/Client/selectors', () => ({
      selectLogin: () => true,
    }));
    test('souldbe navigate to home', () => {
      setTimeout(() => {
        expect(mockNavigate).toHaveBeenCalledWith('/');
      }, 1500);
    });
  });
  describe('login false', () => {
    jest.mock('@containers/Client/selectors', () => ({
      selectLogin: () => false,
    }));
    test('Correct render', () => {
      const { getByTestId } = wrapper;
      expect(getByTestId('reset-password-page')).toBeInTheDocument();
    });

    test('Should be navigate to home', () => {
      const { getByTestId } = wrapper;
      act(() => {
        fireEvent.click(getByTestId('logo'));
      });
      expect(mockNavigate).toHaveBeenCalledWith('/');
    });
    test('Should change field value', () => {
      const inputPassword = screen.getByLabelText('Kata Sandi Baru');
      const inputConfirmPassword = screen.getByLabelText('Konfirmasi Kata Sandi');
      fireEvent.change(inputPassword, { target: { value: 'password' } });
      fireEvent.change(inputConfirmPassword, { target: { value: 'password123' } });
      expect(inputPassword.value).toBe('password');
      expect(inputConfirmPassword.value).toBe('password123');
    });
    test('Should change type pass to text', () => {
      const { getByTestId } = wrapper;
      const button = getByTestId('button-show-pass');
      const inputConfirmPassword = screen.getByLabelText('Konfirmasi Kata Sandi');
      fireEvent.change(button);
      expect(inputConfirmPassword).toHaveAttribute('type', 'password');
    });
    test('Should submit', async () => {
      const { getByTestId } = wrapper;
      const buttonSubmit = getByTestId('button-submit');
      await act(async () => {
        fireEvent.click(buttonSubmit);
      });
    });

    test('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
