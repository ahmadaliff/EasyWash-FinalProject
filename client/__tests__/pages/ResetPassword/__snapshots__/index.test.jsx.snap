// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword Page login false Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="mainWrap"
        data-testid="reset-password-page"
      >
        <div
          class="logoHeader"
          data-testid="logo"
        >
          <img
            alt="logo"
            class="logo"
            src="../../longLogo.svg"
          />
        </div>
        <div
          class="forgotPasswordCard"
        >
          <h2
            class="forgotPasswordHeader"
          >
            Reset kata Sandi
          </h2>
          <form
            class="form"
          >
            <div
              class="MuiFormControl-root inputContainer css-1nrlq1o-MuiFormControl-root"
              data-testid="inputRHF"
            >
              <label
                class="inputLabel"
                for="new_password"
              >
                Kata Sandi Baru
                <input
                  class="input undefined"
                  id="new_password"
                  name="new_password"
                  type="password"
                  value=""
                />
              </label>
            </div>
            <div
              class="MuiFormControl-root inputContainer css-1nrlq1o-MuiFormControl-root"
              data-testid="inputRHF"
            >
              <label
                class="inputLabel"
                for="confirmPassword"
              >
                Konfirmasi Kata Sandi
                <input
                  class="input undefined"
                  id="confirmPassword"
                  name="confirmPassword"
                  type="password"
                  value=""
                />
              </label>
              <label
                class="showPassword"
                for="show"
              >
                <input
                  data-testid="button-show-pass"
                  id="show"
                  name="show"
                  type="checkbox"
                />
                Tampilkan Kata Sandi
              </label>
            </div>
            <button
              class="buttonSubmit"
              data-testid="button-submit"
              disabled=""
              type="submit"
            >
              Reset kata Sandi
            </button>
            <div
              class="formNav"
            >
              <p
                class="nav"
              >
                Punya akun? 
                <b />
              </p>
            </div>
          </form>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="mainWrap"
      data-testid="reset-password-page"
    >
      <div
        class="logoHeader"
        data-testid="logo"
      >
        <img
          alt="logo"
          class="logo"
          src="../../longLogo.svg"
        />
      </div>
      <div
        class="forgotPasswordCard"
      >
        <h2
          class="forgotPasswordHeader"
        >
          Reset kata Sandi
        </h2>
        <form
          class="form"
        >
          <div
            class="MuiFormControl-root inputContainer css-1nrlq1o-MuiFormControl-root"
            data-testid="inputRHF"
          >
            <label
              class="inputLabel"
              for="new_password"
            >
              Kata Sandi Baru
              <input
                class="input undefined"
                id="new_password"
                name="new_password"
                type="password"
                value=""
              />
            </label>
          </div>
          <div
            class="MuiFormControl-root inputContainer css-1nrlq1o-MuiFormControl-root"
            data-testid="inputRHF"
          >
            <label
              class="inputLabel"
              for="confirmPassword"
            >
              Konfirmasi Kata Sandi
              <input
                class="input undefined"
                id="confirmPassword"
                name="confirmPassword"
                type="password"
                value=""
              />
            </label>
            <label
              class="showPassword"
              for="show"
            >
              <input
                data-testid="button-show-pass"
                id="show"
                name="show"
                type="checkbox"
              />
              Tampilkan Kata Sandi
            </label>
          </div>
          <button
            class="buttonSubmit"
            data-testid="button-submit"
            disabled=""
            type="submit"
          >
            Reset kata Sandi
          </button>
          <div
            class="formNav"
          >
            <p
              class="nav"
            >
              Punya akun? 
              <b />
            </p>
          </div>
        </form>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
